src/boot/efi/vmm.c

```diff
@@ -319,6 +339,54 @@ const char* smbios_find_oem_string(const char *name) {
         return NULL;
 }
 
+static const char* smbios_get_string(const SmbiosHeader *header, uint16_t string, uint64_t left) {
+        const char *s = (const void *) header;
+
+        s += header->length;
+        left -= header->length;
```

Comment:
before doing this subtraction, please validate again via an assert that this is not going to underflow, i.e. add `assert(left >= header->length)` before